### ISSUE TITLE
Move convert.yml workflow to metanorma/ci

### ISSUE THREAD

**Issue by CAMOBAP** [metanorma/ci#186]
Also convert this to actions
- https://github.com/metanorma/mn-samples-iec-private/blob/main/Makefile
- https://github.com/metanorma/mn-samples-bsi/blob/main/Makefile

TODO analyze other repos

@ronaldtse let me know if it make sense

**Comment by ronaldtse** [comment]
Makes perfect sense. Please help migrate all known repos to the new workflow. Thanks!

**Comment by ronaldtse** [comment]
Maybe just search for Makefile and the old action?

**Comment by opoudjis** [comment]
Surfacing scope. This ticket — centralising the `convert.yml` build flow currently expressed in `mn-samples-iec-private/Makefile` and `mn-samples-bsi/Makefile` into a reusable `metanorma/ci` workflow — is **separate from the cimas/ci release-chain rehabilitation arc** currently in flight.

The current [rehabilitation arc roadmap](https://github.com/metanorma/cimas/blob/main/plans/cimas-revival-and-release-workflow-realignment.md) covers:

- Release-chain observability + recovery (`#274`, `#302`, `#309`, `cimas#49`)
- cimas-config drift management + design gaps (`#300`)
- Phase B cimas refactor (autoload, OOP/MECE decomposition, YAML schemas)

The mn-samples sample-build centralisation is a **build-system migration** for the sample-document repos rather than a release-chain concern. Worth doing if mn-samples-iec-private / mn-samples-bsi maintenance is currently painful in Makefile form, but doesn't share infrastructure with the release-chain work and won't share an implementation seam.

**Two paths forward:**

1. **Concrete demand signal** — if active mn-samples-bsi/iec-private maintenance is the bottleneck and `convert.yml`-as-reusable-workflow would unblock it, a small PR to define the workflow in `metanorma/ci/.github/workflows/sample-convert.yml` and migrate one of the Makefiles as canary is a 1-2 hour piece of work that doesn't need to wait on the broader rehabilitation.

2. **Defer until Phase B** — if mn-samples is on cruise control, the build-system question can be deferred until Phase B's reusable-workflow rewriting pass naturally absorbs this case.

Leaving open. If the maintenance pain is concrete, ping with the use case and the canary path is small. Otherwise this can ride alongside Phase B.

🤖

**Comment by opoudjis** [comment]
Path 1 implementation ([metanorma/ci#361](https://github.com/metanorma/ci/pull/361) + [mn-samples-bsi#365](https://github.com/metanorma/mn-samples-bsi/pull/365)) declined by @ronaldtse on 2026-07-21 with the reasoning that "2 private repos is not a reason to update a generic workflow" — [see closing comment](https://github.com/metanorma/ci/pull/361#issuecomment-5033717030).

Both PRs closed. This ticket left open — happy to close as won't-fix if the direction has changed permanently, or leave for possible future scope reconsideration.

🤖
